libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -851,7 +851,6 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 	solAssert(_type.isDynamicallySized(), "");
 	solUnimplementedAssert(!_type.isByteArray(), "Byte Arrays not yet implemented!");
 	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "...");
-	solUnimplementedAssert(_type.baseType()->storageSize() == 1, "");
```

Comment:
`recursive_struct_2` was asserting here, but since the function doesn't look like it requires this to be true, I think it was safe to remove the assert. Also, all the test passed for me.